better error message if kid is not found

jkogler-cloudflight opened the pull request:
Closes #19 

I've used a similar error message to what they use here: https://digital.nhs.uk/developer/guides-and-documentation/security-and-authorisation/application-restricted-restful-apis-signed-jwt-authentication

dorinclisu commented:
Thank you for trying to work this out.

Have you tested the changes? Because I have doubts it works correctly. Notice you are raising within a try block, and it's being matched into this generic exception: https://github.com/dorinclisu/fastapi-auth0/blob/541af7f995e350e20409fa0d3625dac41e028e29/src/fastapi_auth0/auth.py#L160-L165 which will result in "Error decoding token" final message.

Please also note than `Auth0UnauthorizedException` is not suitable because it would imply the authentication succeeded yet there are missing permissions. `Auth0UnauthenticatedException` is more suitable for this case.

Please add a test case (I think it's ok to hardcode your token for now, which will surely have a different kid than the one set up in CI) in which you check for the relevant keyword in the message, similar to the other tests.

jkogler-cloudflight commented:
Thanks for the review. 
You're correct. The exception was swallowed by the `except Exception:`.

I've corrected the bug, switched to `Auth0UnauthenticatedException`, tested locally that it works, and also added a test case with an Auth0 access token to a test application of mine. The exception should still be thrown once the token expires.

dorinclisu commented:
Thank you, now it works. I might do a bit more polishing (there's also some commented out code I want to remove) and then publish a new version asap.
